internal/tsc_wrapped/strict_deps.ts

```diff
@@ -87,8 +97,20 @@ export function checkModuleDeps(
     }
     const declFileNames =
         sym.declarations.map(decl => decl.getSourceFile().fileName);
-    if (declFileNames.find(
-            declFileName => !!allowedMap[stripExt(declFileName)])) {
+    const checkDir = (filename: string): boolean => {
+      for (const dir of allowedDirs) {
+        const descendent = PATH_TEST_REGEX.test(path.posix.relative(filename, dir));
```

Comment:
the regex seems kinda overkill, don't you just need `.startsWith('../')`?

Reply:
No because then the rule might descend into a parent. E.g. `../../../parent/no/don't/go/deeper`. Technically all files are connected to each other in some form or another if you go high enough and then descend back down. We want to make sure the file is exclusively a descendant of the allowed directory.